pymatgen/transformations/advanced_transformations.py

```diff
@@ -1360,3 +1361,295 @@ def inverse(self):
+                 structures_per_displacement_val,
+                 floor_displacement=None):
+        """
+        Args:
+            original_structure (Structure): structure to displace, typically
+                already relaxed
+            max_displacement_val (float): maximum displacement value for
+                perturbing the structure (Angstroms)
+            min_displacement_val (float): minimum displacement value for
+                perturbing the structure (Angstroms)
+            num_displacements (int): number of unique displacement values to
+                try, uniformly distributed between 'min_displacement' and
+                'max_displacement'. This argument is ignored if min_displacement
+                is not passed as an argument.
+            structures_per_displacement_val (int): number of perturbed structures to
+                generate for each unique displacement value.
+
+            floor_displacement (Optional float): If None (default), then for a
+                given perturbed structure, all atoms will move the same distance
+                from their original locations. If float, then for a given perturbed
+                structure, the distances that atoms move will be uniformly distributed
+                from a minimum value of 'floor_displacement' to one of the displacement
+                values, 'displacement_val', as generated by 'min_displacement_val',
+                'max_displacement_val', and 'max_displacement'.
+        Returns:
+            List of randomly displaced structures (List of Structures)
+        """
+
+        self.max_disp_val = max_displacement_val
+        self.min_disp_val = min_displacement_val
+        self.num_disps = num_displacements
+        self.structures_per_disp_val = structures_per_displacement_val
+
+        if floor_displacement is not None:
+            self.floor_disp = float(floor_displacement)
+        else:
+            self.floor_disp = None
+
+        self.disp_vals = np.linspace(min_displacement_val, max_displacement_val, num=num_displacements)
+
+    def _random_displacements(self, natom, rmax, rmin, dim=3):
+        #*** Adapted from csld.util.mathtool
+
+        # Generate matrix of size (natom, 3) where each row is a Gaussian random vector
+        # If 'rmin'=None, then all vectors will have magnitude 'rmax'.
+        # Else, magnitudes are uniformly distributed between 'rmin' and 'rmax'.
+
+        dx = np.random.normal(size=(natom, dim)) #Gaussian sampled displacements. Matrix size: (natom, 3)
+        dx_norms = np.linalg.norm(dx, axis=1)
+        veclen = np.full(natom, rmax) if rmin is None else np.random.uniform(rmin, rmax, natom)
+
+        if not 0 in dx_norms:
+            return dx * (veclen / dx_norms)[:, None]
+        else:
+            self._random_displacements(natom, rmax, rmin, dim)
+
+    def _perturb_structure(self, structure, max_disp, floor_disp):
+        # *** Adapted from CSLD's 'polaron_main' file
```

Comment:
the main docstring, e.g. in the """ section should be directly after the function definition. This line can be in the docstring.

You should also give the full `args:` listing of the arguments in the docstring too.